runtime/client_http_request.go

```diff
@@ -188,7 +188,7 @@ func (req *ClientHTTPRequest) Do() (*ClientHTTPResponse, error) {
 	}
 
 	// emit metrics
-	req.metrics.Sent.Inc(1)
+	req.client.ContextMetrics.IncCounter(req.ctx, clientRequest, 1)
```

Comment:
Fixed. It seems the only way to pass contextmetrics to req.metrics is by passing it through req.client.contextMetrics to req.metrics.

Reply:
I'm not sure what that means. `ClientHTTPRequest` in its constructor gets `client` so it can pull out `client.ContextMetrics` into its own instance field. As written it still uses `req.client.ContextMetrics`.

The intention for this change is that if/when we start using Fx or some DI system and change where `ContextMetrics` comes from, we only need to change the constructor rather than all of the call sites.

Reply:
I moved contextMetrics to per request level, please let me know whether it is the correct fix